backend/test/modules/party/party_service_test.py

```diff
@@ -0,0 +1,188 @@
+    assert any(p.id == party.id for p in parties)
+
+
+@pytest.mark.asyncio
+async def test_get_parties_by_contact(party_service: PartyService, sample_party_data: PartyData):
+    party = await party_service.create_party(sample_party_data)
+    
+    # Test contact one
+    parties = await party_service.get_parties_by_contact(sample_party_data.contact_one_id)
+    assert len(parties) >= 1
+    assert any(p.id == party.id for p in parties)
+    
+    # Test contact two
+    parties = await party_service.get_parties_by_contact(sample_party_data.contact_two_id)
+    assert len(parties) >= 1
+    assert any(p.id == party.id for p in parties)
+
+
+@pytest.mark.asyncio
+async def test_get_parties_by_date_range(party_service: PartyService, sample_party_data: PartyData):
+    party = await party_service.create_party(sample_party_data)
+    
+    start_date = sample_party_data.party_datetime - timedelta(hours=1)
+    end_date = sample_party_data.party_datetime + timedelta(hours=1)
+    
+    parties = await party_service.get_parties_by_date_range(start_date, end_date)
+    assert len(parties) >= 1
+    assert any(p.id == party.id for p in parties)
+
+
+@pytest.mark.asyncio
+async def test_update_party(party_service: PartyService, sample_party_data: PartyData):
+    party = await party_service.create_party(sample_party_data)
+    
+    update_data = PartyData(
+        party_datetime=sample_party_data.party_datetime + timedelta(hours=2),
+        address_id=sample_party_data.address_id,
+        contact_one_id=sample_party_data.contact_one_id,
+        contact_two_id=sample_party_data.contact_two_id
+    )
+    
+    updated = await party_service.update_party(party.id, update_data)
+    assert party.id == updated.id
+    assert updated.party_datetime == update_data.party_datetime
+
+
+@pytest.mark.asyncio
+async def test_update_party_not_found(party_service: PartyService, sample_party_data: PartyData):
+    with pytest.raises(PartyNotFoundException):
+        await party_service.update_party(999, sample_party_data)
+
+
+@pytest.mark.asyncio
+async def test_delete_party(party_service: PartyService, sample_party_data: PartyData):
+    party = await party_service.create_party(sample_party_data)
+    deleted = await party_service.delete_party(party.id)
+    assert deleted.id == party.id
+    assert deleted.party_datetime == party.party_datetime
+    
+    with pytest.raises(PartyNotFoundException):
+        await party_service.get_party_by_id(party.id)
+
+
+@pytest.mark.asyncio
+async def test_delete_party_not_found(party_service: PartyService):
+    with pytest.raises(PartyNotFoundException):
+        await party_service.delete_party(999)
+
+
+@pytest.mark.asyncio
+async def test_party_exists(party_service: PartyService, sample_party_data: PartyData):
+    party = await party_service.create_party(sample_party_data)
+    assert await party_service.party_exists(party.id) == True
+    assert await party_service.party_exists(999) == False
```

Comment:
Nit: Getting a linter warning from this. Should be 
```suggestion
    assert await party_service.party_exists(party.id)
    assert not await party_service.party_exists(999)
```
Make sure you have no linter warnings before you push the PR